onnxruntime/core/providers/coreml/builders/impl/base_op_builder.cc

```diff
@@ -50,10 +50,21 @@ Status BaseOpBuilder::AddToModelBuilder(ModelBuilder& model_builder, const Node&
   return Status::OK();
 }
 
-/* static */ std::unique_ptr<COREML_SPEC::NeuralNetworkLayer> BaseOpBuilder::CreateNNLayer(const Node& node) {
-  std::unique_ptr<COREML_SPEC::NeuralNetworkLayer> layer =
-      std::make_unique<COREML_SPEC::NeuralNetworkLayer>();
-  layer->set_name(node.Name());
+/* static */ std::unique_ptr<COREML_SPEC::NeuralNetworkLayer>
+BaseOpBuilder::CreateNNLayer(ModelBuilder& model_builder, const Node& node) {
+  auto layer_name = node.Name();
+  if (layer_name.empty()) {
+    // CoreML requires layer has a name, while the node name is optional in ONNX
+    // In this case, create a unique name for the layer
+    layer_name = model_builder.GetUniqueName(MakeString("Node_", node.Index(), "_type_", node.OpType()));
+  }
```

Comment:
nit: does the string need to be this long or can we just do node.OpType() + node.Index()?

Reply:
It does not have to, just want to make sure it is unique.